packages/salesforce/src/Adaptor.js

```diff
@@ -629,15 +680,19 @@ export function upsert(sObject, externalId, attrs) {
  * })
  * @function
  * @param {boolean} logical - a logical statement that will be evaluated.
- * @param {String} sObject - API name of the sObject.
- * @param {String} externalId - ID.
- * @param {Object} attrs - Field attributes for the new object.
+ * @param {string} sObject - API name of the sObject.
+ * @param {string} externalId - ID.
+ * @param {(object|object[])} attrs - Field attributes for the new object.
  * @returns {Operation}
  */
 export function upsertIf(logical, sObject, externalId, attrs) {
   return state => {
     const resolvedLogical = expandReferences(logical)(state);
 
+    console.warn(
```

Comment:
hmm, I was thinking of something softer than this. Like adding an `@deprecated` tag. I wonder how the docsite will deal with that? I'll check.

We certainly need to add a clear line to the public documentation, even if we can't use @deprecated and even if we keep the logged warning.

Reply:
Hey actually you could add @deprecated, run `pnpm docs:build`, and load the docsite from the `generate-adaptors-devx` branch. That should let you see the changes! That's what I'll be trying this afternoon, but it would be great to test it from your machine

Reply:
@josephjclark This is how the docs look like 
![Screenshot 2024-06-14 at 3 14 38 PM](https://github.com/OpenFn/adaptors/assets/6592749/bedc05d2-186c-432d-b8b9-dad0645be8f9)

Reply:
Hey how are you in dark mode?! I never tried that 😂 Looks neat

Anyway the deprecated thing looks terrible. Hmm. Let's just add a simple string to the docs then

Reply:
It's vscode markdown view 😆 , here is how it looks in docs site 
<img width="954" alt="Screenshot 2024-06-14 at 3 47 15 PM" src="https://github.com/OpenFn/adaptors/assets/6592749/2dccd520-a26e-4032-916e-ae281f906ae5">

Reply:
Is this better ?
<img width="954" alt="Screenshot 2024-06-14 at 3 51 58 PM" src="https://github.com/OpenFn/adaptors/assets/6592749/62807ef9-7ceb-4f17-8d82-f514cc35b871">

Reply:
Better - instead of quoting it, can we make it all bold?

Reply:
And docs.openfn.org org does come with dark mode - I just never switched it on for some reason 
![image](https://github.com/OpenFn/adaptors/assets/7052509/b2363d2f-c931-4e45-a371-765e63f7f7f8)

Reply:
VSC's table stying is way nicer...

Reply:
I have change from quote to bold @josephjclark